sfputil/main.py

```diff
@@ -1564,5 +1603,186 @@ def target(port_name, target):
+
+        physical_port = logical_port_to_physical_port_index(port_name)
+        sfp = platform_chassis.get_sfp(physical_port)
+        if not sfp.get_presence():
+            click.echo("{}: SFP EEPROM not detected\n".format(port_name))
+            sys.exit(EXIT_FAIL)
+
+        try:
+            bytes = bytearray.fromhex(data)
+        except ValueError:
+            click.echo("Error: Data must be a hex string of even length!")
+            sys.exit(EXIT_FAIL)
+
+        from sonic_platform_base.sonic_xcvr.api.public import sff8472
+        api = sfp.get_xcvr_api()
+        if api is None:
+            click.echo('Error: SFP EEPROM not detected!')
+            sys.exit(EXIT_FAIL)
+
+        if not isinstance(api, sff8472.Sff8472Api):
+            overall_offset = get_overall_offset_general(api, page, offset, len(bytes))
+        else:
+            overall_offset = get_overall_offset_sff8472(api, page, offset, len(bytes), wire_addr)
+        success = sfp.write_eeprom(overall_offset, len(bytes), bytes)
+        if not success:
+            click.echo("Error: Failed to write EEPROM!")
+            sys.exit(ERROR_NOT_IMPLEMENTED)
+        if verify:
+            read_data = sfp.read_eeprom(overall_offset, len(bytes))
+            if read_data != bytes:
+                click.echo(f"Error: Write data failed! Write: {''.join('{:02x}'.format(x) for x in bytes)}, read: {''.join('{:02x}'.format(x) for x in read_data)}")
+                sys.exit(EXIT_FAIL)
+    except NotImplementedError:
+        click.echo("This functionality is currently not implemented for this platform")
+        sys.exit(ERROR_NOT_IMPLEMENTED)
+    except ValueError as e:
+        click.echo("Error: {}".format(e))
+        sys.exit(EXIT_FAIL)
+
+
+def get_overall_offset_general(api, page, offset, size):
+    """
+    Validate input parameter page, offset, size and translate them to overall offset
+    Args:
+        api: cable API object
+        page: module EEPROM page number.
+        offset: module EEPROM page offset.
+        size: number bytes of the data to be read/write
+
+    Returns:
+        The overall offset
+    """
+    max_page = 0 if api.is_flat_memory() else MAX_EEPROM_PAGE
+    if max_page == 0 and page != 0:
+        raise ValueError(f'Invalid page number {page}, only page 0 is supported')
+
+    if page < 0 or page > max_page:
+        raise ValueError(f'Invalid page number {page}, valid range: [0, {max_page}]')
+
+    if page == 0:
+        if offset < MIN_OFFSET_FOR_PAGE0 or offset > MAX_EEPROM_OFFSET:
+            raise ValueError(f'Invalid offset {offset} for page 0, valid range: [0, 255]')
+    else:
+        if offset < MIN_OFFSET_FOR_NON_PAGE0 or offset > MAX_EEPROM_OFFSET:
+            raise ValueError(f'Invalid offset {offset} for page {page}, valid range: [128, 255]')
+
+    if size <= 0 or size + offset - 1 > MAX_EEPROM_OFFSET:
+        raise ValueError(f'Invalid size {size}, valid range: [1, {255 - offset + 1}]')
+
```

Comment:
@Junchao-Mellanox please use Click's range option to avoid duplicate code

![image](https://github.com/sonic-net/sonic-utilities/assets/45705344/87431637-500d-4722-a1dc-057b34753b9f)